crates/miden-objects/src/note/note_tag.rs

```diff
@@ -195,27 +205,39 @@ impl NoteTag {
     ///
     /// A note is intended for execution by a single account if the first two bits are zeros
     pub fn is_single_target(&self) -> bool {
-        let first_2_bit = self.0 >> 30;
-        first_2_bit == 0b00
+        match self {
+            NoteTag::NetworkAccount(_) | NoteTag::LocalAccount(_) => true,
+            NoteTag::NetworkUseCase(_, _) | NoteTag::LocalUseCase(_, _) => false,
+        }
     }
```

Comment:
This could just be:

```rust
    pub fn is_single_target(&self) -> bool {
        matches!(self, NoteTag::NetworkAccount(_) | NoteTag::LocalAccount(_))
    }
```